cli/azd/cmd/container.go

```diff
@@ -367,6 +368,25 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		}
 	}
 
+	// Pipelines
+	container.RegisterSingleton(pipeline.NewPipelineManager)
```

Comment:
I am a little nervous about what the container will be resolving to initialize the manager.
Unfortunately, we need to run manual tests for this, as the unit-tests for picking the appropriate provider mocks the input env and args.

Manual tests we need:
- github as default provider when no cmd-args and no azure.yaml config
- github when no cmd-args and github set in azure.yml
- azdo when no cmd-args and azdo set in azure.yml
- github when cmd-args=github (when there is azure.yml config and when there isn't)
- azdo  when cmd-args=azdo (when there is azure.yml config and when there isn't)
- azdo when there is NO cmd-args and no azure.yml config, but previous run used cmd-args=azdo (selection available in .env)

I can find sometime in the future to run these tests manually (if you haven't).  That's the only thing I think could break out of the refactoring,  if the container is not resolving or reading config and/or env at the right time

Reply:
These are the same test cases that are in the pipeline manager unit tests.  The configuration from env, project config, CLI params all come from the container.

I also manually ran through all these test cases and they are working as expected.